src/subscriptions/SubscriptionManager.cpp

```diff
@@ -246,6 +262,14 @@ SubscriptionManager::pubLedger(
         getLedgerPubMessage(lgrInfo, fees, ledgerRange, txnCount)));
 
     ledgerSubscribers_.publish(message);
+
+    if (not bookChangesSubscribers_.empty())
+    {
+        auto const json = RPC::computeBookChanges(lgrInfo, backend_);
+        auto const bookChangesMsg =
+            std::make_shared<Message>(boost::json::serialize(json));
+        bookChangesSubscribers_.publish(bookChangesMsg);
```

Comment:
I think we should add a new function to `SubscriptionManager` called `pubBookChanges`, where we pass the transactions and ledger in as arguments, and then publish. Reason being, we already fetch all of the transactions once, in order to call `pubTransaction`, and it would be more efficient to just use those fetched transactions, instead of fetching the same transactions again to compute the book changes. We fetch the transactions in `ReportingETL::publishLedger`. Fetching transactions is a network call, so it's one of the most expensive operations we perform in the whole application.

Reply:
yea, agree. I'll change this part